### PR TITLE
Feature/tvist1 801 party reference number

### PR DESCRIPTION
**Pull request by jekuaitk**
https://jira.itkdev.dk/browse/TVIST1-801

* Adds reference number to party and using it during hearing mail template generation.

### CODE REVIEW

File: src/Service/MailTemplateHelper.php [hunk 1]
```diff
@@ -322,17 +325,39 @@ private function getValues(?object $entity, TemplateProcessor $templateProcessor
...
+
+                $data += ['relation' => $relationData];
+
+                switch (get_class($entity)) {
+                    case HearingPostRequest::class:
+                        $data['recipient'] += $relationData;
+                        break;
+                    case HearingPostResponse::class:
+                        $data['sender'] += $relationData;
+                }
```
Comment (rimi-itk) [inline src/Service/MailTemplateHelper.php:357]: Add a comment on why we're adding relation data in more than one place.

Consider computing a key to add `$relationData` to, e.g.

```php
$key = $entity instanceof HearingPostRequest ? 'recipient' : 'sender';
$data[$key] += $relationData;
```

File: templates/party/add_party_from_index.html.twig [hunk 2]
```diff
@@ -19,15 +19,25 @@
...
+            <div class="col-md-6 mb-3">
+                {{ form_row(add_party_form.type) }}
+            </div>
+            <div class="col-md-6 mb-3">
+                {{ form_row(add_party_form.referenceNumber) }}
+            </div>
+        </div>
+        <div class="row">
+            <div class="col">
+                <button class="btn btn-success float-end">{{ button_label|default('Add party'|trans([], 'party', 'da')) }}</button>
```
Comment (rimi-itk) [inline templates/party/add_party_from_index.html.twig:38]: The arguments to `trans` are superfluous here.

File: translations/case+intl-icu.da.xlf [hunk 3]
```diff
@@ -1609,6 +1609,10 @@
         <source>Inspection</source>
         <target state="translated">Besigtigelse</target>
       </trans-unit>
+      <trans-unit id="z57ogPt" resname="Reference number">
+        <source>Reference number</source>
+        <target state="translated">Reference nummer</target>
```
Comment (rimi-itk) [inline translations/case+intl-icu.da.xlf:1614]: This should be “Referencenummer” (cf. https://sproget.dk/raad-og-regler/typiske-problemer/et-eller-flere-ord/et-eller-flere-ord/#3dcc7ef8-462b-4015-979b-a35239d24c31).

**Review by rimi-itk — CHANGES_REQUESTED** [review]

File: templates/party/add_party_from_index.html.twig (hunk 2 above)
Comment (jekuaitk) [reply templates/party/add_party_from_index.html.twig:38]: Removed redundant arguments.

File: translations/case+intl-icu.da.xlf (hunk 3 above)
Comment (jekuaitk) [reply translations/case+intl-icu.da.xlf:1614]: Fixed!

File: translations/party+intl-icu.da.xlf [hunk 4]
```diff
@@ -77,6 +77,10 @@
         <source>Identifier</source>
         <target state="translated">Id</target>
       </trans-unit>
+      <trans-unit id="z57ogPt" resname="Reference number">
+        <source>Reference number</source>
+        <target state="translated">Reference nummer</target>
```
[the remark this answers is not in the document]
Comment (jekuaitk) [reply translations/party+intl-icu.da.xlf:82]: Fixed!

File: src/Service/MailTemplateHelper.php (hunk 1 above)
Comment (jekuaitk) [reply src/Service/MailTemplateHelper.php:357]: Good idea, fixed.

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**